src/plugins/intel_npu/src/al/include/intel_npu/al/icompiler.hpp

```diff
@@ -151,6 +151,32 @@ struct NetworkDescription final {
     NetworkMetadata metadata;
 };
 
+/**
+ * @struct CompiledNetwork
+ * @brief Custom container for compiled network, used for export
+ * @var CompiledNetwork::data
+ * Pointer to the address of compiled network
+ * @var CompiledNetwork:size
+ * Size of the compiled network
+ * @var CompiledNetwork::ownedStorage
+ * Plugin owned compiled network storage that is required in case of a driver that
+ * doesn't support graph extension 1.7, as in this case plugin must create a copy of the compiled network.
+ * @note It's unsafe to store either data or size outside of the compiled network object as its destructor
+ * would release the owning container
+ */
+
+struct CompiledNetwork {
+    const uint8_t* data;
+    size_t size;
+    CompiledNetwork(const uint8_t* data, size_t size, const std::vector<uint8_t>&& storage)
```

Comment:
`const std::vector<uint8_t>&&` doesn't make much sense here - you will always get extra copy of the vector, take by non-`const` value